R/ctwas_wrapper.R

```diff
@@ -23,38 +32,97 @@ get_ctwas_meta_data <- function(ld_meta_data_file, regions_table) {
   return(list(LD_info = LD_info, region_info = region_info))
 }
 
-#' Function to generate weights_list from harmonized twas_weights_data for cTWAS multigroup analysis
-#'
-#' @param post_qc_twas_data output from harmonize_twas function for twas results
-#' @return A list of list for weight information for each gene-context pair.
-#' @export
-get_ctwas_weights <- function(post_qc_twas_data, LD_meta_file_path) {
-  chrom <- unique(find_data(post_qc_twas_data, c(2, "chrom")))
-  if (length(chrom) != 1) stop("Data provided contains more than one chromosome. ")
-  genes <- names(post_qc_twas_data)
-  # Compile weights_list
-  weights_list <- list()
-  for (gene in genes) {
-    contexts <- names(post_qc_twas_data[[gene]][["weights_qced"]])
-    for (context in contexts) {
-      standardized_context <- clean_context_names(context, gene)
-      data_type <- post_qc_twas_data[[gene]][["data_type"]][[context]]
-      colnames(post_qc_twas_data[[gene]][["weights_qced"]][[context]]) <- "weight"
-      context_variants <- post_qc_twas_data[[gene]][["variant_names"]][[context]]
-      context_range <- sapply(context_variants, function(variant) as.integer(strsplit(variant, "\\:")[[1]][2]))
-      weights_list[[paste0(gene, "|", data_type, "_", standardized_context)]] <- list(
-        chrom = chrom,
-        p0 = min(context_range),
-        p1 = max(context_range),
-        wgt = post_qc_twas_data[[gene]][["weights_qced"]][[context]],
-        R_wgt = post_qc_twas_data[[gene]]$LD[context_variants, context_variants, drop = FALSE], # ld_list$combined_LD_matrix[context_variants, context_variants],
-        gene_name = gene,
-        weight_name = paste0(data_type, "_", standardized_context),
-        type = data_type,
-        context = standardized_context,
-        n_wgt = length(context_variants)
-      )
+#' Function to extract information from harmonize_twas output as cTWAS input. For each imputable context within a gene,
+#' we select strong variants based on cs set and pip value.
+#' required ctwas package: remotes::install_github("xinhe-lab/ctwas",ref = "multigroup")
+get_ctwas_input <- function(post_qc_data, twas_weights_data, LD_meta_file_path, twas_table, region_block) {
+  get_ctwas_weights <- function(post_qc_twas_data, twas_weights_data) {
+    chrom <- unique(find_data(post_qc_twas_data, c(2, "chrom")))
+    if (length(chrom) != 1) stop("Data provided contains more than one chromosome. ")
+    genes <- names(post_qc_twas_data)
+    # Compile weights_list
+    weights_list <- list()
+    for (gene in genes) {
+      contexts <- names(post_qc_twas_data[[gene]][["weights_qced"]])
+      gene_data <- twas_weights_data[[gene]][["export_twas_weights_db"]]
+      for (context in contexts) {
+        data_type <- post_qc_twas_data[[gene]][["data_type"]][[context]]
+        postqc_scaled_weight <- post_qc_twas_data[[gene]][["weights_qced"]][[context]][["scaled_weights"]]
+        # adjust susie weights for selected number of weight variants
```

Comment:
This step must have already been done in harmonize TWAS weights before you do TWAS? The scaled weights should already have been computed based on the correct SuSiE weights.

Reply:
yes, you're right, this is redundant, and should be removed on the adjust_susie_weights step